scipy/linalg/_decomp_update.pyx

```diff
@@ -0,0 +1,2416 @@
+"""
+Routines for updating QR decompositions
+
+.. versionadded: 0.16.0
+
+"""
+# 
+# Copyright (C) 2014 Eric Moore
+#
+# A few references for Updating QR factorizations:
+# 
+# 1, 2, and 3 cover updates to full decompositons (q is square) and 4 and 5
+# cover updates to thin (economic) decompositions (r is square). Reference 3
+# additonally covers updating complete orthogonal factorizations and cholesky
+# decompositions (i.e. updating R alone).
+#
+# 1. Golub, G. H. & Loan, C. F. van V. Matrix Computations, 3rd Ed.
+#    (Johns Hopkins University Press, 1996).
+#
+# 2. Hammarling, S. & Lucas, C. Updating the QR factorization and the least
+#    squares problem. 1-73 (The University of Manchester, 2008).
+#    at <http://eprints.ma.man.ac.uk/1192/>
+#
+# 3. Gill, P. E., Golub, G. H., Murray, W. & Saunders, M. A. Methods for
+#    modifying matrix factorizations. Math. Comp. 28, 505-535 (1974).
+#
+# 4. Daniel, J. W., Gragg, W. B., Kaufman, L. & Stewart, G. W.
+#    Reorthogonalization and stable algorithms for updating the Gram-Schmidt QR
+#    factorization. Math. Comput. 30, 772-795 (1976).
+#
+# 5. Reichel, L. & Gragg, W. B. Algorithm 686: FORTRAN Subroutines for
+#    Updating the QR Decomposition. ACM Trans. Math. Softw. 16, 369–377 (1990).
+# 
+
+cimport cython
+cimport libc.stdlib
+cimport libc.limits as limits
+from libc.math cimport sqrt, fabs, hypot
+from libc.string cimport memset
+cimport numpy as cnp
+
+cdef int NPY_ANYORDER = 0  
```

Comment:
why define this to zero instead of using ``cnp.NPY_ANYORDER`` in the rest of this file?